File: 2690-house-robber-iv/house-robber-iv.py
class Solution:
    def minCapability(self, nums: List[int], k: int) -> int:
        l, r = min(nums), max(nums)
        while l <= r:
            md = (l + r)//2
            cnt = 0 
            i = 0
            while i < len(nums):
                if nums[i] <= md:
                    cnt += 1
                    i += 1
                i += 1
            if cnt >= k:
                r = md - 1
            else:
                l = md + 1
        return l 

File: 2690-house-robber-iv/house-robber-iv.java
class Solution {
    public int minCapability(int[] a, int k) {
        int l = Integer.MAX_VALUE, r = Integer.MIN_VALUE;
        for(int i : a){
            l = Math.min(l, i);
            r = Math.max(r, i);
        }
        while(l <= r){
            int md = l + (r - l)/2 ;
            int cnt = 0 ;
            for(int i = 0 ; i < a.length ; i ++){
                if(a[i] <= md){
                    cnt ++ ;
                    i ++ ;
                }
            }
            if(cnt >= k){
                r = md - 1;
            }else l = md + 1 ;
        }
        return l ;
    }
}